### ISSUE TITLE
Signed integer overflow in strod.hpp

### ISSUE THREAD

**Issue by pauldreik** [stephenberry/glaze#372]
The following test case causes an integer overflow at [strod.hpp line 677](https://github.com/stephenberry/glaze/blob/9581fb3dee84ec5c6fcfcbcc79c4edae5915a9f3/include/glaze/util/strod.hpp#L677)
```cpp
   "integer overflow in strtod.hpp"_test = [] {
      std::string_view buffer{R"({"i":6E19})"};
      my_struct value{};
      glz::context ctx{};
      [[maybe_unused]] const auto ec = glz::read<glz::opts{.force_conformance = false}>(value, buffer, ctx);
   };
```

The error message is:

> /home/paul/code/glaze/include/glaze/util/strod.hpp:677:17: runtime error: signed integer overflow: 6 * -8446744073709551616 cannot be represented in type 'long'

**Comment by stephenberry** [comment]
Thanks, I opened this issue here as well #368

**Comment by stephenberry** [comment]
Fixed and added unit test with #378
